sdk/src/sysvars/rent.rs

```diff
@@ -153,35 +165,75 @@ impl Rent {
         &*(bytes.as_ptr() as *const Rent)
     }
 
-    /// Calculate how much rent to burn from the collected rent.
+    /// Calculates the minimum balance for rent exemption.
     ///
-    /// The first value returned is the amount burned. The second is the amount
-    /// to distribute to validators.
-    #[inline]
-    pub fn calculate_burn(&self, rent_collected: u64) -> (u64, u64) {
-        let burned_portion = (rent_collected * u64::from(self.burn_percent)) / 100;
-        (burned_portion, rent_collected - burned_portion)
+    /// This method avoids floating-point operations when the `exemption_threshold`
+    /// is the default value.
+    ///
+    /// # Arguments
+    ///
+    /// * `data_len` - The number of bytes in the account
+    ///
+    /// # Returns
+    ///
+    /// The minimum balance in lamports for rent exemption.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `data_len` exceeds the maximum permitted data length.
+    #[deprecated(since = "0.10.0", note = "Use `Rent::try_minimum_balance` instead")]
+    #[inline(always)]
+    pub fn minimum_balance(&self, data_len: usize) -> u64 {
+        self.try_minimum_balance(data_len)
+            .expect("Maximum permitted data length exceeded")
     }
 
-    /// Rent due on account's data length with balance.
-    #[inline]
-    pub fn due(&self, balance: u64, data_len: usize, years_elapsed: f64) -> RentDue {
-        if self.is_exempt(balance, data_len) {
-            RentDue::Exempt
+    /// Calculates the minimum balance for rent exemption without performing
+    /// any validation.
+    ///
+    /// This method avoids floating-point operations when the `exemption_threshold`
+    /// is the default value.
+    ///
+    /// # Arguments
+    ///
+    /// * `data_len` - The number of bytes in the account
+    ///
+    /// # Returns
+    ///
+    /// The minimum balance in lamports for rent exemption.
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure that `data_len` is within permitted the permitted
+    /// limit to avoid overflow.
+    #[inline(always)]
+    pub unsafe fn minimum_balance_unchecked(&self, data_len: usize) -> u64 {
```

Comment:
Is this actually unsafe? If not, I'd say `unchecked` makes it clear enough and remove `unsafe`